### PR TITLE
Update clang-format 10 -> 13

### PR DESCRIPTION
**Pull request by def-**
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test if it works standalone, system.c especially
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/client/components/menus.cpp [hunk 1]
```diff
@@ -2829,9 +2829,10 @@ bool CMenus::HandleListInputs(const CUIRect &View, float &ScrollValue, const flo
 		}
 		if(NewIndex > -1 && NewIndex < NumElems)
 		{
-			//scroll
+			// scroll
 			float IndexY = View.y - ScrollValue * ScrollNum * ElemHeight + NewIndex * ElemHeight;
-			int Scroll = View.y > IndexY ? -1 : View.y + View.h < IndexY + ElemHeight ? 1 : 0;
+			int Scroll = View.y > IndexY ? -1 : View.y + View.h < IndexY + ElemHeight ? 1 :
+                                                                                                    0;
```
Comment (heinrich5991) [inline src/game/client/components/menus.cpp:2835]: Same.

File: src/game/client/components/menus_demo.cpp [hunk 2]
```diff
@@ -676,8 +676,9 @@ CMenus::CListboxItem CMenus::UiDoListboxNextItem(const void *pId, bool Selected,
 					{
 						// scroll
 						float Offset = (NewIndex / gs_ListBoxItemsPerRow - gs_ListBoxNewSelected / gs_ListBoxItemsPerRow) * gs_ListBoxRowHeight;
-						int Scroll = gs_ListBoxOriginalView.y > Item.m_Rect.y + Offset ? -1 :
-														 gs_ListBoxOriginalView.y + gs_ListBoxOriginalView.h < Item.m_Rect.y + Item.m_Rect.h + Offset ? 1 : 0;
+						int Scroll = gs_ListBoxOriginalView.y > Item.m_Rect.y + Offset                                            ? -1 :
+							     gs_ListBoxOriginalView.y + gs_ListBoxOriginalView.h < Item.m_Rect.y + Item.m_Rect.h + Offset ? 1 :
+                                                                                                                                                            0;
```
Comment (heinrich5991) [inline src/game/client/components/menus_demo.cpp:681]: Weird indentation.

File: src/game/client/components/menus_demo.cpp [hunk 3]
```diff
@@ -807,8 +808,9 @@ void CMenus::DemolistOnUpdate(bool Reset)
 			m_DemolistSelectedIndex = SelectedIndex;
 	}
 
-	m_DemolistSelectedIndex = Reset ? m_lDemos.size() > 0 ? 0 : -1 :
-					  m_DemolistSelectedIndex >= m_lDemos.size() ? m_lDemos.size() - 1 : m_DemolistSelectedIndex;
+	m_DemolistSelectedIndex = Reset                                      ? m_lDemos.size() > 0 ? 0 : -1 :
+				  m_DemolistSelectedIndex >= m_lDemos.size() ? m_lDemos.size() - 1 :
+                                                                               m_DemolistSelectedIndex;
```
Comment (heinrich5991) [inline src/game/client/components/menus_demo.cpp:813]: Same.

**Review by heinrich5991 — CHANGES_REQUESTED** [review]
Except for the one comment issue, this seems to make ternary operator expressions (objectively?) worse.

Maybe `AlignOperands` from https://releases.llvm.org/13.0.0/tools/clang/docs/ClangFormatStyleOptions.html needs to be changed to something else? It's a new thing from clang-format 12, apparently.

File: src/base/system.h [hunk 4]
```diff
@@ -1428,7 +1428,7 @@ int str_utf8_dist_buffer(const char *a, const char *b, int *buf, int buf_len);
 		b_len - Length of the second string.
 		buf - Buffer for the function.
 		buf_len - Length of the buffer, must be at least as long as
-		          the length of both strings combined plus two.
+			  the length of both strings combined plus two.
```
[the remark this answers is not in the document]
Comment (def-) [reply src/base/system.h:1431]: True, done

**Comment by def-** [comment]
I can't find a good solution for the formatting problems, github workflow doesn't support ubuntu 21.10 yet

**Pull request closed without merging**